### PR TITLE
deprecated Parser.parseToQuantityValue()

### PR DESCRIPTION
**Pull request by franklbh**
Deprecating Parser.parseToQuantityValue(), which will be marked as deprecated in version 4.10. The existing method Parser.parseQuantityString() should be used as a replacement.

**Reason for Deprecation:**
The method Parser.parseQuantityString() is more appropriate for handling quantity parsing and is already in use in multiple places, such as:

- In presentation-components: UseQuantityValueInput (via ParserSpec).
- In core-frontend: Through quantityFormatter in tools like Accudraw.
- In ecschema-metadata: Used in bearing tests.
- In imodel-components-react: QuantityInput and QuantityNumberInput.

The method Parser.parseToQuantityValue() is used sparingly in:

- Appui's test.
- core-frontend's test.

Given that the system has already moved to using Parser.parseQuantityString() in most cases, removing Parser.parseToQuantityValue() is reasonable.

### CODE REVIEW

**Comment by hl662** [comment]
The replacement for the method you are deprecating already exists, it would not be something new introduced into 4.10. Do you instead mean Parser.parseToQuantityValue() method will be deprecated in 4.10? Please update your PR description to reflect this.

**Comment by hl662** [comment]
Your PR CI build [failed](https://github.com/iTwin/itwinjs-core/actions/runs/10964614337/job/30448607499?pr=7180), you missed running `rush extract-api`. Refer to our root [README.md](https://github.com/iTwin/itwinjs-core/actions/runs/10964614337/job/30448607499?pr=7180) on instructions for fixing it.

**Review by rschili — APPROVED** [review]

**Review by aruniverse — COMMENTED** [review]
whats up with all the white space changes?

**Comment by franklbh** [comment]
> whats up with all the white space changes?

I think the spaces were automatically added, now they're gone

**Comment by aruniverse** [comment]
> > whats up with all the white space changes?
> 
> I think the spaces were automatically added, now they're gone

please check again, 
image

also, lets add a nextver.md update about the deprecation

File: core/frontend/src/quantity-formatting/QuantityFormatter.ts [hunk 1]
```diff
@@ -554,7 +554,7 @@ export class QuantityFormatter implements UnitsProvider {
     try {
       // force all cached data to be reinitialized
       await IModelApp.quantityFormatter.onInitialized();
-    } catch (err) {
```
Comment (hl662) [inline core/frontend/src/quantity-formatting/QuantityFormatter.ts]: revert this

File: core/frontend/src/test/BearingQuantityType.ts [hunk 2]
```diff
@@ -119,7 +119,6 @@ class BearingParserSpec extends ParserSpec {
     const parsedRadians = Parser.parseQuantityString(adjustedString, this);
     if (Parser.isParsedQuantity(parsedRadians)) {
       if (prefix === "N" && suffix === "E") {
-
```
Comment (hl662) [inline core/frontend/src/test/BearingQuantityType.ts]: revert this too

**Review by hl662 — APPROVED** [review]

File: common/api/ecschema-editing.api.md
Comment (hl662) [inline common/api/ecschema-editing.api.md:1]: revert this too

File: common/changes/@itwin/core-frontend/Frank-deprecation-quantity_2024-09-24-21-17.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@itwin/core-frontend",
+      "comment": "Refactored parseToQuantityValue to parseQuantityString in core frontend tests",
```
Comment (pmconne) [inline common/changes/@itwin/core-frontend/Frank-deprecation-quantity_2024-09-24-21-17.json:5]:
```suggestion
      "comment": "",
```

Change logs are for users. Users don't care about tests.

File: common/changes/@itwin/core-quantity/Frank-deprecation-quantity_2024-10-07-15-26.json
Comment (hl662) [inline common/changes/@itwin/core-quantity/Frank-deprecation-quantity_2024-10-07-15-26.json:1]: why is there another change file added? You have an existing one for your changes already

**Review by pmconne — APPROVED** [review]

**Review by ben-polinsky — APPROVED** [review]

**Review by ColinKerr — APPROVED** [review]

**Pull request merged**